WallPaperDev/WallPaperDev/Views/Controllers/CreateGoalViewController.swift

```diff
@@ -56,39 +60,35 @@ extension CreateGoalViewController {
   private func setupButton() {
     createGoalView.createButton.addTarget(self, action: #selector(createTapped), for: .touchUpInside)
   }
+
+  func presentError() {
+    let alertView = UIAlertController(title: Localized.string("invalid"),
+                                      message: Localized.string("create_goal_error_message"),
+                                      preferredStyle: .alert)
+    let action = UIAlertAction(title: Localized.string("alert_ok"), style: .cancel, handler: nil)
+    alertView.addAction(action)
+    present(alertView, animated: true, completion: nil)
+  }
 }
 
 // MARK: - Objc functions
-
 extension CreateGoalViewController {
   @objc func createTapped() {
-    guard let userGoalName = createGoalView.goalNameTextField.text,
-      let userGoalSummary = createGoalView.goalDescriptionTextView.text else {
-      return
-    }
-
-    func presentError() {
-      let alertView = UIAlertController(title: Localized.string("invalid"),
-                                        message: Localized.string("create_goal_error_message"),
-                                        preferredStyle: .alert)
-      let action = UIAlertAction(title: Localized.string("ok_action"),
-                                 style: .cancel,
-                                 handler: nil)
-      alertView.addAction(action)
-      self.present(alertView, animated: true, completion: nil)
+    if validInputs() {
+      guard let userGoalName = createGoalView.goalNameTextField.text,
+        let userGoalSummary = createGoalView.goalDescriptionTextView.text else {
+        return
+      }
+      coreDataStack.createGoal(userGoalName, userGoalSummary)
+      navigationController?.popViewController(animated: true)
+    } else {
+      presentError()
     }
-    coreDataStack.createGoal(userGoalName, userGoalSummary)
-    navigationController?.popViewController(animated: true)
-  }
-
-  func presentError() {
-    let alertView = UIAlertController(title: "Invalid", message: "You can't create a goal without a name and description", preferredStyle: .alert)
-    let action = UIAlertAction(title: "Ok", style: .cancel, handler: nil)
-    alertView.addAction(action)
-    present(alertView, animated: true, completion: nil)
   }
 }
 
+// MARK: - UITextViewDelegate
+// TODO: There is a bug here
```

Comment:
:) Let's add some details as to what the bug is! It is amazing how easy it is to forget bugs a week or two later after stepping away from the codebase

Reply:
Oh sorry I forgot to remove the comment 😅. The bug was the logic for input validation and I fixed it in this PR.